### PR TITLE
fix!: update tf fmt to 1.1.X; enforce lint and docs 

### PR DESCRIPTION
**Pull request by edulop91**
### Summary
- Check docs doesn't fail if docs out out date, just warns. We should fail so we catch these in CI. 
- fix terraform fmt
- fix terraform lint
- CI improvements
- fix some failing tests
- get us up to TF 1.1.0 syntax + fmt

### CODE REVIEW

**Review by jacobrheath — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -8,43 +7,46 @@ jobs:
...
-          echo STEP_ID=$(echo script | sum | cut -f 1 -d \ ) >> $GITHUB_ENV
-      - name: Run tests
-        env:
-          REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.REVIEWDOG_GITHUB_API_TOKEN }}
-        run: |
-          buildevents cmd $TRACE_ID $STEP_ID codecov-validate -- curl --data-binary @codecov.yml https://codecov.io/validate
-          buildevents cmd $TRACE_ID $STEP_ID make-lint-ci -- make lint-ci
-          buildevents cmd $TRACE_ID $STEP_ID make-test-ci -- make test-ci
-          buildevents step $TRACE_ID $STEP_ID $STEP_START script
+      - name: markdownlint
```
Comment (edulop91) [inline .github/workflows/build.yml:45]: use existing reviewdog actions now to simplify things

File: apply/apply_test.go [hunk 2]
```diff
@@ -269,14 +269,14 @@ func TestApplyModuleInvocation(t *testing.T) {
 	r.Nil(e)
 	i, e := afero.ReadFile(fs, "mymodule/main.tf")
 	r.Nil(e)
-	expected := "# Auto-generated by fogg. Do not edit\n# Make improvements in fogg, so that everyone can benefit.\n\nmodule test-module {\n  source = \"../test-module\"\n  bar    = local.bar\n  foo    = local.foo\n\n}\n"
+	expected := "# Auto-generated by fogg. Do not edit\n# Make improvements in fogg, so that everyone can benefit.\n\nmodule \"test-module\" {\n  source = \"../test-module\"\n  bar    = local.bar\n  foo    = local.foo\n\n}\n"
```
Comment (edulop91) [inline apply/apply_test.go:272]: fix tests to be terraform fmt compatible

File: apply/golden_file_test.go [hunk 3]
```diff
@@ -33,6 +33,7 @@ func TestIntegration(t *testing.T) {
 		{"github_actions"},
 		{"circleci"},
 		{"tfe_provider_yaml"},
+		{"remote_backend_yaml"},
```
Comment (edulop91) [inline apply/golden_file_test.go:36]: add this to the codegen test suite

File: apply/test-module/outputs.tf [hunk 4]
```diff
@@ -1,7 +1,7 @@
 output "foo" {
-  value = "${var.foo}"
+  value = var.foo
```
Comment (edulop91) [inline apply/test-module/outputs.tf:2]: terraform fmt

File: templates/templates/common/auth0_provider.tmpl [hunk 5]
```diff
@@ -1,5 +1,5 @@
 {{ define "auth0_provider" }}
-provider auth0 {
+provider "auth0" {
```
Comment (edulop91) [inline templates/templates/common/auth0_provider.tmpl:2]: all these templates for terraform fmt compat

File: templates/templates/repo/scripts/component.mk [hunk 6]
```diff
@@ -33,12 +33,7 @@ endif
 .PHONY: lint-tflint
 
 lint-terraform-fmt: terraform ## run `terraform fmt` in check mode
-	@printf "fmt check: "
-	@for f in $(TF); do \
-		printf . \
-		$(terraform_command) fmt $(TF_ARGS) --check=true --diff=true $$f || exit $$? ; \
-	done
-	@echo
+	$(terraform_command) fmt $(TF_ARGS) --check=true --diff=true
```
Comment (edulop91) [inline templates/templates/repo/scripts/component.mk:36]: actually fmt our files...

File: templates/templates/repo/scripts/module.mk [hunk 7]
```diff
@@ -9,9 +9,7 @@ all: fmt lint doc
 .PHONY: all
 
 fmt: terraform ## run terraform fmt on this module
-	@printf "fmt: ";
-	@for f in $(TF); do printf .; $(terraform_command) fmt $(TF_ARGS) $$f; done
-	@echo
+	@$(terraform_command) fmt $(TF_ARGS)
```
Comment (edulop91) [inline templates/templates/repo/scripts/module.mk:12]: actually fmt

File: templates/templates/repo/scripts/module.mk [hunk 8]
```diff
@@ -21,14 +19,10 @@ lint: lint-tf ## run all linters on this module
 .PHONY: lint
 
 lint-tf: terraform ## run terraform linters on this module
-	@printf "fmt: ";
-	@for f in $(TF); do \
-	  printf .; $(terraform_command) fmt $(TF_ARGS) --check=true --diff=true || exit $$? ; \
-	done
-	@echo
+	$(terraform_command) fmt $(TF_ARGS) --check=true --diff=true
```
Comment (edulop91) [inline templates/templates/repo/scripts/module.mk:22]: actually lint

File: templates/templates/repo/scripts/module.mk [hunk 9]
```diff
@@ -39,6 +33,7 @@ check-docs: ## check that this module's docs are up-to-date
 	@bash $(REPO_ROOT)/scripts/update-readme.sh check; \
 	if [ ! $$? -eq 0 ];  then \
 		echo "Docs are out of date, run \`make docs\`"; \
+		exit 1 ; \
```
Comment (edulop91) [inline templates/templates/repo/scripts/module.mk:36]: fail if readmes not up to date

File: testdata/auth0_provider_yaml/scripts/component.mk [hunk 10]
```diff
@@ -33,12 +33,7 @@ endif
 .PHONY: lint-tflint
 
 lint-terraform-fmt: terraform ## run `terraform fmt` in check mode
-	@printf "fmt check: "
-	@for f in $(TF); do \
-		printf . \
-		$(terraform_command) fmt $(TF_ARGS) --check=true --diff=true $$f || exit $$? ; \
-	done
-	@echo
+	$(terraform_command) fmt $(TF_ARGS) --check=true --diff=true
```
Comment (edulop91) [inline testdata/auth0_provider_yaml/scripts/component.mk:36]: testdata/* is codegenerated tests

**Pull request merged**